File: snippets/python-async/connectors-id-oauth-authorize-post.py
import cohere
import asyncio

co = cohere.AsyncClient("<<apiKey>>")


async def main():
    response = await co.connectors.o_auth_authorize(
        connector_id="test-id", after_token_redirect="https://test.com"
    )
    print(response)

asyncio.run(main())

File: snippets/snippets/java/app/src/main/java/ConnectorsIdOauthAuthorizePost.java
/* (C)2024 */
import com.cohere.api.Cohere;
import com.cohere.api.resources.connectors.requests.ConnectorsOAuthAuthorizeRequest;
import com.cohere.api.types.OAuthAuthorizeResponse;

public class ConnectorsIdOauthAuthorizePost {
  public static void main(String[] args) {
    Cohere cohere = Cohere.builder().clientName("snippet").build();

    OAuthAuthorizeResponse response =
        cohere
            .connectors()
            .oAuthAuthorize(
                "test-id",
                ConnectorsOAuthAuthorizeRequest.builder()
                    .afterTokenRedirect("https://connector-example.com/search")
                    .build());

    System.out.println(response);
  }
}
